flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java

```diff
@@ -349,7 +349,12 @@ public int numKeyValueStatesByName() {
     }
 
     // TODO remove this once heap-based timers are working with RocksDB incremental snapshots!
-    public boolean requiresLegacySynchronousTimerSnapshots(CheckpointType checkpointOptions) {
+    public boolean requiresLegacySynchronousTimerSnapshots(CheckpointType checkpointType) {
+        return false;
+    }
+
+    // TODO remove this once heap-based timers are working with RocksDB incremental snapshots!
```

Comment:
I think this method still useful even `requiresLegacySynchronousTimerSnapshots` could be removed, heap state backend should return false for this method. And we could remove the comment.

Reply:
Which method do you mean?
(I'm not removing any methods in this PR)

Reply:
I mean the `isStateImmutableInStateBackend` method. I think this comment description dose not match the role which `isStateImmutableInStateBackend` is playing. In other words, we cannot remove `isStateImmutableInStateBackend` even heap-based timers could work with RocksDB incremental snapshots.

Reply:
Oh, I see. Yes, I agree and will remove the comment.
Thanks!